scripts/create-component/plop-templates/src/components/{{componentName}}/{{componentName}}.types.ts.hbs

```diff
@@ -4,16 +4,13 @@ export type {{componentName}}Slots = {
   root: Slot<'div'>;
 };
 
-type {{componentName}}Commons = {
-  // TODO Add things shared between props and state here
-};
-
 /**
  * {{componentName}} Props
  */
-export type {{componentName}}Props = ComponentProps<{{componentName}}Slots> & {{componentName}}Commons;
+export type {{componentName}}Props = ComponentProps<{{componentName}}Slots> & {};
 
 /**
  * State used in rendering {{componentName}}
  */
-export type {{componentName}}State = ComponentState<{{componentName}}Slots> & {{componentName}}Commons;
+ // TODO: Replace never below with union of props to pick from {{componentName}}Props above.
+export type {{componentName}}State = ComponentState<{{componentName}}Slots> & Required<Pick<{{componentName}}Props, never>>;
```

Comment:
this isn't the best DX as with this user will get probably type errors and might spent some additional time figuring out what's wrong.

can we go with something like following ?

```ts
export type {{componentName}}State = ComponentState<{{componentName}}Slots> 
// TODO: Uncomment following section and provide union of props to pick from {{componentName}}Props.
// & Required<Pick<{{componentName}}Props, 'propName'>>
;
```

Reply:
Sure, I can go with that too. I've updated it!